nipype/algorithms/confounds.py

```diff
@@ -513,12 +509,15 @@ def _list_outputs(self):
             outputs['detrended_file'] = op.abspath(self.inputs.detrended_file)
         return outputs
 
-def regress_poly(degree, data):
+def regress_poly(degree, data, remove_mean=False, axis=-1):
     ''' returns data with degree polynomial regressed out.
-    The last dimension (i.e. data.shape[-1]) should be time.
+    Be default it is calculated along the last axis (usu. time)
     '''
     datashape = data.shape
-    timepoints = datashape[-1]
+    timepoints = datashape[axis]
+
+    if remove_mean: # i.e. regress_poly degree 0, which the following does not do
```

Comment:
```
if remove_mean:
    datahat = X[:, 1:].dot(betas[1:, ...]).T
else
    datahat = X.dot(betas).T
```

also in this function `if remove_mean and degree == 0` should raise an error - they are equivalently the same thing.